src/metabase/api/email.clj

```diff
@@ -31,9 +31,16 @@
                                 :email-smtp-port "Wrong host or port"}}
           creds-error {:errors {:email-smtp-username "Wrong username or password"
                                 :email-smtp-password "Wrong username or password"}}
-          message     (str/join ": " (map ex-message (u/full-exception-chain error)))]
-          (log/warn "Problem connecting to mail server:" message)
-      (condp re-find message
+          exceptions  (u/full-exception-chain error)
+          message     (str/join ": " (map ex-message exceptions))
+          check       (fn check [regex|exception-class [m es]]
```

Comment:
I have nothing against using pipes in symbol names but I have no idea what this is supposed to mean. Is `regex|exception-class` a regex pattern that you pipe the name of `exception-class` into to see if it matches something? That's my best guess. Maybe this is worth a comment

Reply:
Oh I get it now. It means `regex-or-exception-class`. I only figured it out by reading the code which I think is a sign that it's not the greatest variable name. I'm generally a fan of fancy characters in symbol names because we can -- I've put `₁` and `°` in names in the codebase in the past but eventually removed them after other people whined about them -- but I think we should make sure their meaning is obvious.

I think `x|y` is a neat convention but if this is the only place we're using it it's not much of a convention. So maybe we should hate fun and just call this `regex-or-exception-class`. Just so no one else gets super confused.

Reply:
Other observation: `check` is a little bit terse. Check _what_? `m` and `es` are cryptic. Is `m` supposed to be a message? 

I probably sound like a broken record but code is write-once, read-many so making these variable names a little longer makes the code a lot easier to understand

Reply:
ha sorry. I've seen that as a standard convention in the past. I'll rename it so its not confusing.

Reply:
updated. thanks for the pointers.